operators/bosh-operator/BoshStaggeredDeploymentPoller.js

```diff
@@ -26,7 +26,18 @@ class BoshStaggeredDeploymentPoller extends BaseStatusPoller {
     const deploymentName = _.get(resourceBody, 'status.response.deployment_name');
     return DirectorService
       .createInstance(instanceId, resourceOptions)
-      .then(directorService => directorService.createOrUpdateDeployment(deploymentName, resourceOptions))
+      .then(directorService => {
+        if (deploymentName) {
```

Comment:
This multiple if/else seems a bit confusing at first, only to later figure out that if deploymentName exists then you dont have to initialize in case of create/update. How about changing the lines 30 - 38 like this:
```
resourceOptions._deploymentName = deploymentName;
directorService.update(resourceOptions);
```
You just update deployment name in resourceOptions and invoke just update and Changing DirectoService.initialize method like the below. Initialize will additionally take in second arg (payload), whatever is sent to create/update and if deployment exists, it straightaway initialises network segment index.
```
initialize(operation, params) {
    return Promise
      .try(() => {
        if(params._deploymentName) {
           return this.getNetworkSegmentIndex(deploymentName);
        }
        this.operation = operation.type;
        if (operation.type === CONST.OPERATION_TYPE.CREATE) {
          return this.acquireNetworkSegmentIndex(this.guid);
        }
        return this.findNetworkSegmentIndex(this.guid);
      })
      .tap(networkSegmentIndex => {
        assert.ok(_.isInteger(networkSegmentIndex), `Network segment index '${networkSegmentIndex}' must be an integer`);
        this.networkSegmentIndex = networkSegmentIndex;
      })
      .tap(() => {
        if (operation.type === 'delete') {
          return Promise
            .all([
              this.platformManager.preInstanceDeleteOperations({
                guid: this.guid
              }),
              this.deleteRestoreFile()
            ]);
        }
      });
  }
```

Reply:
did the same thing bit different way. But can discuss if needs to be changed.